### PR TITLE
Refresh tasks and offers every time the tabs are visible

### PR DESCRIPTION
**Pull request by berryve**

### CODE REVIEW

File: app/src/main/java/org/kinecosystem/kinit/model/earn/Task.kt [hunk 1]
```diff
@@ -29,12 +29,14 @@ data class Task(
     val provider: Provider? = null,
     @SerializedName("type")
     val type: String? = null,
+    @SerializedName("updated_at")
```
Comment (shayb) [inline app/src/main/java/org/kinecosystem/kinit/model/earn/Task.kt]: no sure why is it part of the task model

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/EarnViewModel.kt [hunk 2]
```diff
@@ -153,13 +141,34 @@ class EarnViewModel(val taskRepository: TasksRepository, val wallet: Wallet,
...
+
+        taskService.retrieveNextTask(object : OperationCompletionCallback {
+            override fun onError(errorCode: Int) {
+            }
+
+            override fun onSuccess() {
+                if (taskRepository.task != null) {
+                    taskRepository.task?.lastUpdatedAt?.let {
+                        if (it > currentLastUpdatedTime ||
+                            currentTaskDisplayedId != taskRepository.task?.id.orEmpty()) {
```
Comment (shayb) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/EarnViewModel.kt]: why do you need this check?

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/EarnViewModel.kt [hunk 3]
```diff
@@ -153,13 +141,34 @@ class EarnViewModel(val taskRepository: TasksRepository, val wallet: Wallet,
...
+        val currentTaskDisplayedId = taskRepository.task?.id.orEmpty()
+
+        taskService.retrieveNextTask(object : OperationCompletionCallback {
+            override fun onError(errorCode: Int) {
+            }
+
+            override fun onSuccess() {
+                if (taskRepository.task != null) {
+                    taskRepository.task?.lastUpdatedAt?.let {
+                        if (it > currentLastUpdatedTime ||
```
Comment (shayb) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/EarnViewModel.kt]: when will it be false?

Comment (berryve) [reply app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/EarnViewModel.kt]: I am only going to refresh the screen if something has really changed (either this is a completely new task or some of the details of the task have been changed and therefore the last updated time has been changed)

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/EarnViewModel.kt (hunk 2 above)
Comment (berryve) [reply app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/EarnViewModel.kt]: same answer

**Pull request merged**